### PR TITLE
test: batch 1 remove strings testid app code

### LINKED ISSUE #7865: [Batch 1/7] Remove all instances of strings in TestID property in the app code
## Description
We have instances where testIDs are not consistently stored in variables. This lack of uniformity can lead to redundant testID declarations throughout the codebase, resulting in code that is harder to maintain and prone to errors.

For instance, consider the following code snippet: `testID={'request-qrcode-button'}.` This line directly specifies a testID value within the code, and similar declarations may be scattered throughout the codebase.

A more effective and best-practice approach is to store the testID value as a variable and then reference that variable when defining the testID property. By doing so, we centralize the testID values, reducing redundancy and making it easier to manage and maintain these identifiers throughout the codebase. This approach enhances code consistency and reduces the likelihood of introducing errors, ultimately contributing to a more robust and maintainable codebase.

## Technical details:

Replace all string definitions used for testIDs. For example: `testID={BrowserUrlModalSelectorsIDs.CONTAINER}`
All variables should be group by the file where they're being call
All variables should be group in a JSON
File structure should follow a similar structure as the main app structure

## Acceptance Criteria 

- [ ] pr_regression_e2e_pipeline and pr_smoke_e2e_pipeline in bitrise should have passing tests

- [ ] These files should no longer selector strings in the TestID prop:
app/component-library/components/Icons/Icon/Icon.test.tsx
app/components/UI/AccountOverview/index.js
app/components/UI/AccountRightButton/index.tsx
app/components/UI/AddCustomCollectible/index.tsx
app/components/UI/AddressInputs/index.js
app/components/UI/ApproveTransactionReview/index.js
app/components/UI/AssetList/index.js

### PR DESCRIPTION
**Pull request by SamuelSalas**
## **Description**
Replace all string definitions used for testIDs. These files should no longer selector strings in the TestID prop:

- app/component-library/components/Icons/Icon/Icon.test.tsx
- app/components/UI/AccountOverview/index.js
- app/components/UI/AccountRightButton/index.tsx
- app/components/UI/AddCustomCollectible/index.tsx
- app/components/UI/AddressInputs/index.js
- app/components/UI/ApproveTransactionReview/index.js
- app/components/UI/AssetList/index.js

## **Related issues**

Fixes: 7865

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

- Regression test run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/17d88968-f5ee-4f2c-956f-b0d731e43226
- Smoke test run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/3ad78d3e-e3f9-4b10-9cf4-493755dd3ae8

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/AddAsset/__snapshots__/AddAsset.test.tsx.snap [hunk 1]
```diff
@@ -315,7 +315,7 @@ exports[`AddAsset component renders correctly 1`] = `
                   null,
                 ]
               }
-              testID="add-custom-asset-cancel-button"
+              testID=""
```
Comment (cortisiko) [inline app/components/Views/AddAsset/__snapshots__/AddAsset.test.tsx.snap:318]: hmmm. Question: did yarn `run test:unit -u `output this? Seems a bit strange to just have an empty string for a testID

File: app/components/UI/AssetList/index.js [hunk 2]
```diff
@@ -81,7 +80,7 @@ export default class AssetList extends PureComponent {
               containerStyle={styles.item}
               onPress={() => handleSelectAsset(searchResults[i])} // eslint-disable-line
               key={i}
-              {...generateTestId(Platform, TOKEN_RESULTS_LIST_ID)}
+              testID={TOKEN_RESULTS_LIST_ID}
```
Comment (cortisiko) [inline app/components/UI/AssetList/index.js:83]: small win: Can you import the selector from the respective selector object in `e2e/selectors`

File: app/components/UI/ApproveTransactionReview/index.js [hunk 3]
```diff
@@ -469,7 +469,7 @@ class ApproveTransactionReview extends PureComponent {
       },
     );
     if (isMultiLayerFeeNetwork(chainId)) {
-      this.fetchEstimatedL1Fee();
+      await this.fetchEstimatedL1Fee();
```
Comment (cortisiko) [inline app/components/UI/ApproveTransactionReview/index.js:472]: hmmm is the `await` needed?

File: app/components/UI/ApproveTransactionReview/index.js [hunk 4]
```diff
@@ -314,7 +314,7 @@ class ApproveTransactionReview extends PureComponent {
       const eth = new Eth(
         Engine.context.NetworkController.getProviderAndBlockTracker().provider,
       );
-      const result = await fetchEstimatedMultiLayerL1Fee(eth, {
+      const result = fetchEstimatedMultiLayerL1Fee(eth, {
```
Comment (cortisiko) [inline app/components/UI/ApproveTransactionReview/index.js:317]: any reason why we remove the `await`

**Review by cortisiko — CHANGES_REQUESTED** [review]
Left a few comments, please run regression and smoke.

File: app/components/Views/AddAsset/__snapshots__/AddAsset.test.tsx.snap (hunk 1 above)
Comment (SamuelSalas) [reply app/components/Views/AddAsset/__snapshots__/AddAsset.test.tsx.snap:318]: Yes, I ran `run test:unit -u` and that was the result.

File: app/components/UI/ApproveTransactionReview/index.js (hunk 3 above)
Comment (SamuelSalas) [reply app/components/UI/ApproveTransactionReview/index.js:472]: Recommended by my IDE. I can remove it.

File: app/components/UI/ApproveTransactionReview/index.js (hunk 4 above)
Comment (cortisiko) [reply app/components/UI/ApproveTransactionReview/index.js:317]: yeah please restore

File: app/components/UI/ApproveTransactionReview/index.js (hunk 3 above)
Comment (cortisiko) [reply app/components/UI/ApproveTransactionReview/index.js:472]: yeah please restore

File: app/components/UI/AssetList/index.js (hunk 2 above)
Comment (SamuelSalas) [reply app/components/UI/AssetList/index.js]: Done

**Review by cortisiko — APPROVED** [review]

File: app/components/UI/ApproveTransactionReview/index.js (hunk 4 above)
Comment (SamuelSalas) [reply app/components/UI/ApproveTransactionReview/index.js]: done

File: app/components/UI/ApproveTransactionReview/index.js (hunk 3 above)
Comment (SamuelSalas) [reply app/components/UI/ApproveTransactionReview/index.js]: done

**Pull request merged**